cmd/joy/release.go

```diff
@@ -384,17 +385,25 @@ func NewGitCommands() *cobra.Command {
 					return fmt.Errorf("getting target tag from release: %w", err)
 				}
 
-				gitargs := append([]string{"git", command}, args[1:]...)
-				gitargs = append(gitargs, sourceTag+".."+targetTag)
+				gitArgs := append([]string{"git", command}, args[1:]...)
+				gitArgs = append(gitArgs, sourceTag+".."+targetTag)
 
-				fmt.Println(color.InCyan("running:"), color.InBold(strings.Join(gitargs, " ")))
+				if project.Spec.RepositorySubpaths != nil {
+					if !slices.Contains(gitArgs, "--") {
+						gitArgs = append(gitArgs, "--")
+					}
+					gitArgs = append(gitArgs, project.Spec.RepositorySubpaths...)
```

Comment:
Why do we want a project to have multiple subpaths?

Doesn't it it make for sense to have multiple projects with the same repo but different subpaths?

IE:
```
project: applicartions -> nestoca/public-apps + /applications
project: documents    -> nestoca/pulbic-apps + /documents
...
```

This way we only log the changes related to the subpath related to the project?

I may be misunderstanding either how are projects are setup or how git log -- paths... work?

Reply:
Projects changes are much more than just the `app/<app-name>`

See https://github.com/nestoca/catalog/pull/692/files?w=1#diff-997462567ee0af560610519be390296da8a66b6907d044d59a4f78962287c2dbR10-R13